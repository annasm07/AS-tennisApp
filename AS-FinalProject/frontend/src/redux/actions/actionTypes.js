export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
