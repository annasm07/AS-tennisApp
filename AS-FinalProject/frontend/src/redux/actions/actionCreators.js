import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
