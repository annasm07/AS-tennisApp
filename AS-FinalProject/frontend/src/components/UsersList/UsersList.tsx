import React, {useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {getAllPlayers} from '../../redux/actions/actionCreators';
import {Text, StyleSheet, View, Image, ScrollView} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';

function UsersList() {
  let tokens = useSelector((store: any) => store.tokens);
  let players = useSelector((store: any) => store.players);
  let player = useSelector((store: any) => store.player);
  const dispatch = useDispatch();

  const newPlayersArray = players.filter(
    (playerInList: any) => playerInList._id !== player._id,
  );

  useEffect(() => {
    dispatch(getAllPlayers(tokens[0]));
  }, [dispatch, tokens]);

  return (
    <SafeAreaView>
      <ScrollView style={styles.scroll}>
        <Text style={styles.title}>Find Other Users</Text>
        {players.length ? (
          newPlayersArray.map((playerToPrint: any) => (
            <View style={styles.playerBox}>
              <Image
                style={styles.playerImage}
                source={{
                  uri: `${playerToPrint.img}`,
                }}
              />
              <View>
                <Text style={styles.names} key={playerToPrint._id}>
                  {playerToPrint.name}
                </Text>
              </View>
            </View>
          ))
        ) : (
          <Image
            source={{
              uri: 'https://www.gatoslechuzos.com/static/loading.gif',
            }}
            style={styles.loading}
          />
        )}
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  scroll: {
    paddingTop: 30,
    marginBottom: 90,
  },
  playerBox: {
    backgroundColor: '#E5E5E5',
    borderRadius: 10,
    padding: 10,
    margin: 15,
    flexDirection: 'row',
  },
  playerImage: {
    width: 60,
    height: 60,
    borderRadius: 100,
  },
  title: {
    fontSize: 26,
    paddingLeft: 20,
    marginTop: -25,
  },
  names: {
    fontSize: 20,
    paddingLeft: 30,
    marginTop: 10,
    textTransform: 'capitalize',
  },
  loading: {
    width: 200,
    height: 200,
    top: '45%',
    left: '40%',
  },
});

export default UsersList;
