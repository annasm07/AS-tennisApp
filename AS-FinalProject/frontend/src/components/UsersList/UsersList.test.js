import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import UsersList from './UsersList';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a UsersList component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    player: {
      _id: 'abc456789',
    },
    tokens: ['abc', 'cde'],
    players: [
      {
        _id: 'abc456789',
        img: 'dfghjkl',
        name: 'ASM',
      },
      {
        _id: 'rtyu67890',
        img: 'dfghjkl',
        name: 'hjk',
      },
    ],
  });
  beforeEach(() => {
    jest.spyOn(actions, 'getPlayerInfo').mockReturnValueOnce({type: ''});
  });

  test('Should render UsersList component', () => {
    const usersList = render(
      <Provider store={store}>
        <UsersList />
      </Provider>,
    );
    expect(usersList).toMatchSnapshot();
  });
});

describe('When invoked a UsersList component players.length === undefined', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    player: {
      _id: 'abc456789',
    },
    tokens: ['abc', 'cde'],
    players: [],
  });
  beforeEach(() => {
    jest.spyOn(actions, 'getAllPlayers').mockReturnValueOnce({type: ''});
  });

  test('Should render UsersList component', () => {
    const usersList = render(
      <Provider store={store}>
        <UsersList />
      </Provider>,
    );
    expect(usersList).toMatchSnapshot();
  });
});
