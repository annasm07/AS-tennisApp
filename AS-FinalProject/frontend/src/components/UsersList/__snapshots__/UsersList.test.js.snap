// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a UsersList component players.length === undefined Should render UsersList component 1`] = `
<RNCSafeAreaView>
  <RCTScrollView
    style={
      Object {
        "marginBottom": 90,
        "paddingTop": 30,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 26,
            "marginTop": -25,
            "paddingLeft": 20,
          }
        }
      >
        Find Other Users
      </Text>
      <Image
        source={
          Object {
            "uri": "https://www.gatoslechuzos.com/static/loading.gif",
          }
        }
        style={
          Object {
            "height": 200,
            "left": "40%",
            "top": "45%",
            "width": 200,
          }
        }
      />
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
